appserver/ejb/ejb-http-remoting/client/src/test/java/fish/payara/ejb/http/client/adapter/example/stub/NoOpJmsStub.java

```diff
@@ -0,0 +1,746 @@
+/*
+ *    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ *    Copyright (c) [2019] Payara Foundation and/or its affiliates. All rights reserved.
+ *
+ *    The contents of this file are subject to the terms of either the GNU
+ *    General Public License Version 2 only ("GPL") or the Common Development
+ *    and Distribution License("CDDL") (collectively, the "License").  You
+ *    may not use this file except in compliance with the License.  You can
+ *    obtain a copy of the License at
+ *    https://github.com/payara/Payara/blob/master/LICENSE.txt
+ *    See the License for the specific
+ *    language governing permissions and limitations under the License.
+ *
+ *    When distributing the software, include this License Header Notice in each
+ *    file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ *    GPL Classpath Exception:
+ *    The Payara Foundation designates this particular file as subject to the "Classpath"
+ *    exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ *    file that accompanied this code.
+ *
+ *    Modifications:
+ *    If applicable, add the following below the License Header, with the fields
+ *    enclosed by brackets [] replaced by your own identifying information:
+ *    "Portions Copyright [year] [name of copyright owner]"
+ *
+ *    Contributor(s):
+ *    If you wish your version of this file to be governed by only the CDDL or
+ *    only the GPL Version 2, indicate your decision by adding "[Contributor]
+ *    elects to include this software in this distribution under the [CDDL or GPL
+ *    Version 2] license."  If you don't indicate a single choice of license, a
+ *    recipient has the option to distribute your version of this file under
+ *    either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *    its licensees as provided above.  However, if you add GPL Version 2 code
+ *    and therefore, elected the GPL Version 2 license, then the option applies
+ *    only if the new code is made subject to such option by the copyright
+ *    holder.
+ */
+
+package fish.payara.ejb.http.client.adapter.example.stub;
+
+import javax.jms.*;
+import java.io.Serializable;
+import java.util.Enumeration;
+
+/**
+ * A stub that throws UnsupportedOperationException or JMSException for any operation (except for close).
+ * Basis for any more useful stub
+ */
+public class NoOpJmsStub {
+    private NoOpJmsStub() {
+
+    }
+
+    static JMSException jmsException() {
+        return new JMSException("This operation is not supported by this stub");
+    }
+
+    static JMSRuntimeException jmsRuntime() {
+        return new JMSRuntimeException("This operation is not supported by this stub");
+    }
+
+    static class NoOpConnectionFactory implements ConnectionFactory {
```

Comment:
I wonder if there is a way using mocks that does the same but is mostly a one-liner per interface.

Reply:
Yes, Mockito would work perfectly here, I just wanted to make example in plain Java. I had to reinforce this decision to myself every hundred lines :D